users/views.py

```diff
@@ -68,11 +76,206 @@ def get_serializer_class(self):
 
 class UserViewSet(viewsets.ModelViewSet):
     queryset = CustomUser.objects.all()
-    permission_classes = (IsOwnerOrAdmin,)
+    permission_classes = (AuthenticatedAdminOrUser,)
 
     def get_serializer_class(self):
         if self.action == Action.RETRIEVE.value:
             return UserDetailSerializer
         if self.action == Action.UPDATE.value:
             return UserUpdateSerializer
         return UserSerializer
+
+
+@login_required
+def index(request):
+    # Number of visits to this view, as counted in the session variable.
+    num_visits = request.session.get('num_visits', 0)
+    request.session['num_visits'] = num_visits+1
+
+    # Render the HTML template index.html with the data in the context variable.
+    return render(
+        request,
+        'home.html',
+        context={
+            'num_visits': num_visits},  # num_visits appended
+    )
```

Comment:
@kbeker  
This feature is just to make the home page not so empty - a form of curiosity. The input counter is saved in the user's cookie. I only added it temporarily...hoping that it could be removed later. Should I remove it or improve like you said?

Reply:
Ok, so leave it now as it is, but please add task to pivotal or github to add visit counter to database. @Karrp @MartynaAnnaGottschling

Reply:
Done